src/inverted-index.js

```diff
@@ -62,15 +86,14 @@ class InvertedIndex {
     // This object stores the index of the current document
     let index = {};
     // Ensure JSON file is not empty or invalid
-    if (docToIndex.length === 0) {
-      this.indexes[filename] = 'JSON file is empty';
-    } else if (!docToIndex[0].title && !docToIndex[0].text) {
-      this.indexes[filename] = 'JSON file is invalid';
+    if (InvertedIndex.validateFile(docToIndex) === 'Empty file') {
+      this.indexes[filename] = ['JSON file is empty'];
+    } else if (InvertedIndex.validateFile(docToIndex) === 'Invalid file') {
+      this.indexes[filename] = ['JSON file is invalid'];
     } else {
       docToIndex.forEach((document, docIndex) => {
```

Comment:
### Nitpick

`document` is a keyword, like `window`. You shouldn't use keywords for your variables.